include/RAJA/pattern/kernel/Lambda.hpp

```diff
@@ -30,16 +30,51 @@
 #include "RAJA/util/macros.hpp"
 #include "RAJA/util/types.hpp"
 
-#include "RAJA/pattern/kernel/ArgHelper.hpp"
 #include "RAJA/pattern/kernel/internal.hpp"
 
 namespace RAJA
 {
 
+namespace internal
+{
+struct lambda_arg_seg_t
+{};
+
+struct lambda_arg_param_t
+{};
+
+struct lambda_arg_offset_t
+{};
+
+template<typename T>
+struct lambda_arg_value_t
+{
+    using type = T;
+};
+
+template<typename T, camp::idx_t>
+struct LambdaArg
+{
+};
+
+}
+
 namespace statement
 {
 
 
+template<camp::idx_t ... args>
```

Comment:
I'm not sure if it's worth reworking, but having the element be a LambdaArg and take a NTTP int as the second param means these can't be used directly with the camp map primitives.  We could probably make some adapters or something for that if it would be useful, but for now that only works with lists of lists of at least two type elements each, first element as key.

Reply:
could you explain why this is an issue?  was there some refactoring with the camp map primatives that you wanted to do here?

It would be easy to provide an alias that expands the first argument of Lambda<L,...> to Lambda2<LambdaArg<L,>, ...>.   Is that what you mean?

Reply:
It's not necessarily an issue, but if it were a `list<list<arg_t, num_t>...>` you could manipulate these with the list and map operations directly.  Whether that matters or not is another question, if they're just tags it probably doesn't, but if they need lookups and similar it might be good.